eventbus.go

```diff
@@ -383,114 +292,89 @@ func createFilter(topic string, m eh.EventMatcher) []string {
+	}
 
-	// if retryCount exceeds number of delays, use the last defined delay value
-	if int(retryCount) >= len(b.queueDelays) {
-		ttl = b.queueDelays[len(b.queueDelays)-1]
-	} else {
-		ttl = b.queueDelays[retryCount]
+	if b.consumeConn, err = setupConnection(b.addr, b.logger,
+		clarimq.WithConnectionOptionConnectionName(fmt.Sprintf("%s_consume_connection", b.appID)),
+		clarimq.WithConnectionOptionMultipleLoggers(b.loggers),
+	); err != nil {
+		return fmt.Errorf(errMessage, err)
 	}
 
-	headers := rabbitmq.Table(delivery.Headers)
-	headers[retryCountHeader] = retryCount + 1
-
-	err := b.publisher.Publish(
-		delivery.Body,
-		[]string{dlQueuePrefix + "_" + ttl.String()},
-		rabbitmq.WithPublishOptionsContentType("application/json"),
-		rabbitmq.WithPublishOptionsMandatory,
-		rabbitmq.WithPublishOptionsPersistentDelivery,
-		rabbitmq.WithPublishOptionsExchange(dlxName),
-		rabbitmq.WithPublishOptionsHeaders(headers),
-	)
-	if err != nil {
-		return rabbitmq.NackRequeue, fmt.Errorf("failed to publish dead letter message: %w", err)
+	if b.publisher, err = clarimq.NewPublisher(b.publishConn); err != nil {
+		return fmt.Errorf(errMessage, err)
 	}
 
-	return rabbitmq.Ack, nil
+	return nil
 }
 
-func (b *EventBus) declareDLX(dlxName, dlQueuePrefix, requeueRoutingKey string) error {
-	connection, err := amqp091.Dial(b.addr)
-	if err != nil {
-		return fmt.Errorf("failed to connect to amqp091: %w", err)
-	}
+func setupConnection(uri string, logger *logger, options ...clarimq.ConnectionOption) (*clarimq.Connection, error) {
+	const errMessage = "failed to setup connection: %w"
 
-	channel, err := connection.Channel()
+	conn, err := clarimq.NewConnection(uri, options...)
 	if err != nil {
-		return fmt.Errorf("failed to open amqp091 channel: %w", err)
+		return nil, fmt.Errorf(errMessage, err)
 	}
 
-	// declare a single dead letter exchange
-	err = channel.ExchangeDeclare(
-		dlxName,  // name
-		"direct", // kind
-		true,     // durable
-		false,    // autoDelete
-		false,    // internal
-		false,    // noWait
-		nil,      // args
-	)
-	if err != nil {
-		return fmt.Errorf("failed to declare amqp091 dead letter exchange: %w", err)
-	}
+	go func() {
+		for err := range conn.NotifyAutoRecoveryFail() {
+			if err == nil {
+				return
+			}
 
-	// declare and bind queues with ttl values
-	for _, ttl := range b.queueDelays {
-		queueName := dlQueuePrefix + "_" + ttl.String()
-
-		_, err = channel.QueueDeclare(
-			queueName, // name
-			true,      // durable
-			false,     // autoDelete
-			false,     // exclusive
-			false,     // noWait
-			map[string]interface{}{
-				"x-dead-letter-exchange":    b.exchangeName, // original exchange, in which the event gets retried
-				"x-dead-letter-routing-key": requeueRoutingKey,
-				"x-message-ttl":             ttl.Milliseconds(),
-			}, // args
-		)
-		if err != nil {
-			return fmt.Errorf("failed to declare amqp091 dead letter queue: %w", err)
+			logger.logWarn("recovery failed", "error", err)
```

Comment:
should this maybe be more than a warn? 😄 
should this also maybe run into the error channel of the eventbus?